### ISSUE TITLE
plot(section,which="Rrho") error with smooth.spline

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1905]
I discovered this in testing for #1878, and will fix it in a commit that addresses about a half-dozen recent issues.  I want not just a clean build and a passing of the test suite, but also a plot of *everything*, before I start closing this web of issues.

``` r
library(oce)
#> Loading required package: gsw
data(section)
plot(section, which="Rrho")
#> Error in smooth.spline(p[ok], temperature[ok], df = df): need at least four unique 'x' values
```

<sup>Created on 2022-01-01 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by dankelley** [comment]
Related:
``` r
library(oce)
#> Loading required package: gsw
data(section)
plot(section, which="Rrho")
#> Error in smooth.spline(p[ok], temperature[ok], df = df): need at least four unique 'x' values
section[["station",1]][["Rrho"]]
#> Warning in smooth.spline(p[ok], temperature[ok], df = df): not using invalid df;
#> must have 1 < df <= n := #{unique x} = 5
#> Warning in smooth.spline(p[ok], salinity[ok], df = df): not using invalid df;
#> must have 1 < df <= n := #{unique x} = 5
#> Warning in smooth.spline(p[ok], theta[ok], df = df): not using invalid df; must
#> have 1 < df <= n := #{unique x} = 5
#> [1] 0.1280710 0.1886879 0.8558694 0.4485078 0.2768589
```

<sup>Created on 2022-01-01 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by dankelley** [comment]
This problem has been solved in "develop" branch, commit baf04c064d3631fa002ccb6b3dab3acbba31574b. 

The reprex below is also available at https://github.com/dankelley/oce-issues/blob/main/19xx/1905/1905a.R;  ignore the confusing contours, but focus on the fact that this works silently, sans error and sans warning.

``` r
library(oce)
#> Loading required package: gsw
library(testthat)
data(section)
expect_silent(plot(section, which="Rrho"))
```

![](https://i.imgur.com/2uNH7oZ.png)

<sup>Created on 2022-01-02 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>
